internal/waitext/error.go

```diff
@@ -0,0 +1,22 @@
+package waitext
+
+// NonTransientError is used in conjuction with Retry or RetryWithContext
+// to allow the operation to signal that retrying is pointless
+// It works the same as PermanentError in github.com/cenkalti/backoff
+type NonTransientError struct {
```

Comment:
Please consider minimizing the amount of abstraction we introduce. We could just make retry function return `(error, bool)`. The 2nd bool indicates whether the function should be retried or not.